python/OmniFoldTTbar.py

```diff
@@ -263,23 +264,40 @@ def _get_input_arrays(self, preprocess=True):
         # truth level
         arr_gen = self.handle_sig.get_arrays(self.varnames_truth, valid_only=False)
 
-        if preprocess:
-            logger.info("Preprocess feature arrays")
+        # preprocess input arrays
 
-            # estimate the order of magnitude
-            logger.debug("Divide each variable by its order of magnitude")
-            # use only the valid events
-            xmean_reco = np.mean(np.abs(self.handle_obs[self.varnames_reco]), axis=0)
-            xoom_reco = 10**(np.log10(xmean_reco).astype(int))
-            arr_data /= xoom_reco
-            arr_sim /= xoom_reco
+        p = preprocessor.get()
 
-            xmean_truth = np.mean(np.abs(self.handle_sig[self.varnames_truth]), axis=0)
-            xoom_truth = 10**(np.log10(xmean_truth).astype(int))
-            arr_gen /= xoom_truth
+        arr_data = p.preprocess(self.varnames_reco, arr_data)
+        # TODO: a truth known option?
```

Comment:
Just want to confirm this is not needed. Since the preprocessed feature arrays are only used in training, I think it is fine to normalize / standardize arr_data alone independent of the truth. We only get the weights out of training so this step shouldn't affect stuff like Chi^2 calculation at all.

Reply:
Yep